### DISCUSSION TITLE
标点调整 Punctuation

### DISCUSSION THREAD

**Discussion by sena-nana** [sena-nana/basedpyright#6] [Ideas]
如题

**Comment by NCBM** [comment]
# 引号

目前对于引号的处理大致有以下几种：

1. 左、右弯引号，视为半角符号（即需要额外补充空格）
2. 左、右弯引号，视为全角符号（即不需额外补充空格）
3. 直角引号
4. 英文直引号
5. 英文左引号
6. ~~单、双尖括号~~

**Reply by sena-nana** [reply]
目前的翻译中以 2 为标准，即中文语境中最常用的符号。
直角引号为书面符号，具有独特的观感，但实际很少有人使用，更多出现在日文中，因此并未采用直角引号。

**Reply by NCBM** [reply]
注意到如文本项 `Diagnostic.argMissingForParams` 实际提供文本中存在大概率为硬编码的直引号。

![屏幕截图_20240421_213742](https://github.com/sena-nana/basedpyright/assets/37037264/3b18e47d-3a42-4b3e-9367-e699e0421cba)

~~还有英文逗号~~

**Reply by NCBM** [reply]
目前大致可以确认一部分硬编码直引号的代码。
![屏幕截图_20240421_223821](https://github.com/sena-nana/basedpyright/assets/37037264/85acf56b-7fed-4939-b30d-d8829987c932)

**Reply by sena-nana** [reply]
寄，考虑到统一性就得都改成直引号了

**Reply by NCBM** [reply]
如果不追求完全的统一性的话，可以只对代码元素使用直引号。

**Reply by sena-nana** [reply]
查了下按照《夹用英文的中文文本的标点符号用法》，应当使用中文引号（
可以先按照代码元素使用半角英文引号，中文内容使用中文引号试一下。等完成以后找几个人试用看看感受

**Reply by NCBM** [reply]
好像大多数使用引号的地方都是代码元素（

**Reply by sena-nana** [reply]
那就统一吧（

**Reply by NCBM** [reply]
注意到 basedpyright 独有的文本项中开始使用了英文左引号 `，可以考虑是否继续讨论。

**Reply by sena-nana** [reply]
> 注意到 basedpyright 独有的文本项中开始使用了英文左引号 `，可以考虑是否继续讨论。

有error lens插件渲染效果的截图么

**Comment by NCBM** [comment]
> > 注意到 basedpyright 独有的文本项中开始使用了英文左引号 `，可以考虑是否继续讨论。
> 
> 有error lens插件渲染效果的截图么

我这边还没找到复现 basedpyright 中此类诊断信息的代码，不过个人认为呈现效果应该与下图（ruff 的诊断信息）类似：

![屏幕截图_20240426_112140](https://github.com/sena-nana/basedpyright/assets/37037264/2fef3ade-b558-4f14-a17f-5188ee2b6371)

**Reply by sena-nana** [reply]
我认为可以尝试，目前看效果应该是和英文引号类似的，但是有强调这部分是代码的作用
